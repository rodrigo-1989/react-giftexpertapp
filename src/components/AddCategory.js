import React,{ useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategorie  } ) => {
    const [inputValue,setInputValue ]=useState(''); 
    
    const handleInputChange = (e)=>{
           setInputValue( e.target.value );      
    }
    const handleSubmit =(e) =>{
        e.preventDefault();
        if (inputValue.trim().length > 2){
            setCategorie(cats => [inputValue,...cats]);
            setInputValue( '' ); }
    }
    return (
        <form onSubmit= { handleSubmit }>
            <input type="text"
                value= { inputValue }
                onChange= { handleInputChange }/>
            
        </form>
    )
}
AddCategory.propTypes = {
    setCategorie: PropTypes.func.isRequired
}
