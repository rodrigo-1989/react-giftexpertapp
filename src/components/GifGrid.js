import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';
export const GifGrid = ({ category }) => { 

       const { data:images } =  useFetchGifs( category );
    return (
        <>
        <h3>{ category }</h3>
       
            <div className="card-grid"> 
                    {
                    images.map( img =>(
                        <GifGridItem key = { img.id }  { ...img }/>
                    ))
                    }
            </div>
        </>
    )
}
 
GifGrid.propTypes={
    category: PropTypes.string.isRequired
}