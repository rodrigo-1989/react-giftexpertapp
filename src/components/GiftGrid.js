import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'

 import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({ category }) => { 

       const { data:images } =  useFetchGifs( category );
    return (
        <>
        <h3>{ category }</h3>
       
            <div className="card-grid"> 
                    {
                    images.map( img =>(
                        <GiftGridItem key = { img.id }  { ...img }/>
                    ))
                    }
            </div>
        </>
    )
}
 