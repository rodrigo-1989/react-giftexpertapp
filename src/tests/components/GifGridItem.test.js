import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import React from 'react';

describe('Pruevas en GifGrid Item', () => {
    const title = 'Un titulo';
    const url = 'https://localhots/algo.jpg';
    const wraper = shallow( <GifGridItem title = { title } url = { url }/> ); 

    test('Debe de mostrar el componenete correctamente', () => {

        expect ( wraper ).toMatchSnapshot();
        
    })
    test('Debe de tener un parrafo con el titulo', () => {
        const p  =  wraper.find('p');

        expect(p.text().trim()).toBe( title ) 
    })
    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img = wraper.find( 'img' );
        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title  );
    })
    test('Debe tener animate_fadeIn', () => {
        const div = wraper.find( 'div' );
        //console.log(div.prop('className'));
        const className = div.prop('className');
        expect( className.includes('') ).toBe( true );
        
    })
    
    
    
})

 