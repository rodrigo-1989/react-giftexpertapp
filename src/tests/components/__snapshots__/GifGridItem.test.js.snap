// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en GifGrid Item Debe de mostrar el componenete correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Un titulo"
    src="https://localhots/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
